File: scratch/analyze_logo.js
import { Jimp } from 'jimp';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

async function main() {
  const imgPath = path.join(__dirname, '..', 'public', 'images', 'LOGO.png');
  console.log('Loading image:', imgPath);
  
  const image = await Jimp.read(imgPath);
  const width = image.bitmap.width;
  const height = image.bitmap.height;
  
  console.log(`Image dimensions: ${width}x${height}`);
  
  // Let's look at a 10x10 grid of pixels at the top-left
  console.log('Top-left 10x10 pixel colors:');
  for (let y = 0; y < 10; y++) {
    let row = [];
    for (let x = 0; x < 10; x++) {
      const color = image.getPixelColor(x, y);
      // Get RGBA channels
      const r = (color >> 24) & 0xff;
      const g = (color >> 16) & 0xff;
      const b = (color >> 8) & 0xff;
      const a = color & 0xff;
      row.push(`#${r.toString(16).padStart(2,'0')}${g.toString(16).padStart(2,'0')}${b.toString(16).padStart(2,'0')} (${a})`);
    }
    console.log(`Row ${y}:`, row.slice(0, 5).join(' | '));
  }
}

main().catch(console.error);

File: scratch/analyze_logo.py
import os
from PIL import Image

def analyze():
    img_path = "../public/images/LOGO.png"
    if not os.path.exists(img_path):
        img_path = "public/images/LOGO.png"
    
    img = Image.open(img_path)
    print("Format:", img.format)
    print("Size:", img.size)
    print("Mode:", img.mode)
    
    # Check some pixels
    w, h = img.size
    print("Top-left pixel:", img.getpixel((0, 0)))
    print("Top-right pixel:", img.getpixel((w - 1, 0)))
    print("Bottom-left pixel:", img.getpixel((0, h - 1)))
    print("Bottom-right pixel:", img.getpixel((w - 1, h - 1)))
    
    # Get standard colors in a 10x10 grid on the left/top edges
    for y in range(10):
        row = [img.getpixel((x, y)) for x in range(10)]
        print(f"Row {y}:", [p[:3] if len(p) > 3 else p for p in row])

if __name__ == "__main__":
    analyze()
